syntax = "proto3";

package simulation;

service SimulationService {
  rpc Simulate (SimulateRequest) returns (SimulateResponse) {}
}

message SimulateRequest {
  string code = 1;
  string lab = 2;
  float duration = 3;
  float distance = 4;
}

message SimulateResponse {
  string status = 1;
  string lab_label = 2;
  InputSettings input_settings = 3;
  repeated LogEntry logs = 4;
  string server_stderr = 5;
  string error = 6;
  string details = 7;
  float duration = 8;
}

message InputSettings {
  string lab = 1;
  float duration = 2;
  float distance = 3;
}

message LogEntry {
  float time = 1;
  int32 pin = 2;
  int32 level = 3;
}
